### PR TITLE
fix(plasma-core): improve type-coverage

### PR DESCRIPTION
**Pull request by Yeti-or**
:baby_chick: Download canary assets:

[colors--canary.231.3539470148.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.231.3539470148.xml)  
[color_sbermarket_ios-swift--canary.231.3539470148.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.231.3539470148.swift)  
[color_sbermarket_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.231.3539470148.kt)  
[color_sbermarket_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.231.3539470148.ts)  
[color_sberprime_ios-swift--canary.231.3539470148.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.231.3539470148.swift)  
[color_sberprime_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.231.3539470148.kt)  
[color_sberprime_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.231.3539470148.ts)  
[PlasmaTokensColor--canary.231.3539470148.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.231.3539470148.swift)  
[typo_mage_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.231.3539470148.kt)  
[typo_mage_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.231.3539470148.ts)  
[typo_plasma_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.231.3539470148.kt)  
[typo_plasma_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.231.3539470148.ts)  
[typo_ruler_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.231.3539470148.kt)  
[typo_ruler_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.231.3539470148.ts)  
[typo_sage_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.231.3539470148.kt)  
[typo_sage_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.231.3539470148.ts)  
[typo_sbermarket_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.231.3539470148.kt)  
[typo_sbermarket_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.231.3539470148.ts)  
[typo_soulmate_kotlin--canary.231.3539470148.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.231.3539470148.kt)  
[typo_soulmate_react-native--canary.231.3539470148.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.231.3539470148.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.115.1-canary.231.3539470148.0
  npm install @salutejs/plasma-core@1.82.1-canary.231.3539470148.0
  npm install @salutejs/plasma-icons@1.109.1-canary.231.3539470148.0
  npm install @salutejs/plasma-temple@1.113.1-canary.231.3539470148.0
  npm install @salutejs/plasma-ui@1.146.1-canary.231.3539470148.0
  npm install @salutejs/plasma-web@1.143.1-canary.231.3539470148.0
  npm install @salutejs/plasma-sb-utils@0.80.1-canary.231.3539470148.0
  # or 
  yarn add @salutejs/plasma-b2c@1.115.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-core@1.82.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-icons@1.109.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-temple@1.113.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-ui@1.146.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-web@1.143.1-canary.231.3539470148.0
  yarn add @salutejs/plasma-sb-utils@0.80.1-canary.231.3539470148.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-231/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-231/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-231/
  ui storybook: http://plasma.sberdevices.ru/pr-231/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-231/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-231/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-231/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-231/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/temple-storybook/

  </details>

File: packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const StyledTextSkeleton = styled.div`
     width: 100%;
 `;
 
+// TODO: https://github.com/salute-developers/plasma/issues/230
 export interface TextSkeletonProps extends TextSizeProps, Partial<RoundnessProps> {
     /**
      * Количетво линий скелета
```
Comment (Yakutoc) [inline packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx:21]: [typo]
Количетво -> Количество

File: packages/plasma-core/src/types/ThemeProviderContextBase.ts [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
     /**
      * Отключить любую анимацию компонента.
      */
     lowPerformance?: boolean;
 
     deviceScale?: number;
 }
+
+// TODO: https://github.com/salute-developers/plasma/issues/232
+// declare module 'styled-components' {
```
Comment (Yakutoc) [inline packages/plasma-core/src/types/ThemeProviderContextBase.ts:13]: @Yeti-or Закомментированный код нужен?

File: packages/plasma-core/src/utils/extractTextFrom.ts [hunk 3]
```diff
@@ -8,12 +8,12 @@ export const extractTextFrom = (textSource?: string | number | null | ReactNode)
         case 'number':
             return textSource.toString();
         case 'object': {
-            if (!isValidElement(textSource) || !textSource.props || !textSource.props.children) {
+            if (!isValidElement(textSource) || (!textSource.props as any) || !textSource.props.children) {
```
Comment (Yakutoc) [inline packages/plasma-core/src/utils/extractTextFrom.ts:11]: @Yeti-or У нас есть возможность обойти использование `any`?

File: packages/plasma-core/src/types/ThemeProviderContextBase.ts (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-core/src/types/ThemeProviderContextBase.ts:13]: Это одно из двух возможных решений, в целом не нужен, оставлю просто issue

File: packages/plasma-core/src/utils/extractTextFrom.ts (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-core/src/utils/extractTextFrom.ts:11]: конкретно тут у меня не вышло, на самом деле уберу

File: packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx:21]: https://github.com/salute-developers/plasma/pull/241

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-231/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-231/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-231/
  ui storybook: http://plasma.sberdevices.ru/pr-231/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-231/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-231/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-231/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-231/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-231/temple-storybook/

  </details>

File: packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx [hunk 4]
```diff
@@ -38,10 +39,10 @@ export const TextSkeleton: React.FC<TextSkeletonProps & React.HTMLAttributes<HTM
     ...props
 }) => (
     <StyledTextSkeleton {...props}>
-        {Array.from(Array(lines), (_, i) => {
-            let w;
+        {Array.from(Array<number>(lines), (_, i) => {
+            let w: number;
             if (width) {
-                w = width;
+                w = width as number;
```
Comment (neretin-trike) [inline packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx:45]: давай вместо явного приведения сделаем такую проверку:

```
 if (typeof width === 'number') {
```

File: packages/plasma-core/src/utils/extractTextFrom.ts [hunk 5]
```diff
@@ -11,9 +11,9 @@ export const extractTextFrom = (textSource?: string | number | null | ReactNode)
             if (!isValidElement(textSource) || !textSource.props || !textSource.props.children) {
                 return '';
             }
-            return Children.map(textSource.props.children, (child) => {
+            return (Children.map(textSource.props.children, (child) => {
                 return extractTextFrom(child);
-            }).join('');
+            }) as Array<string>).join('');
```
Comment (neretin-trike) [inline packages/plasma-core/src/utils/extractTextFrom.ts:16]: мож тут лучше?
```
 Children.map<string, ReactChild>
```

File: packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx (hunk 4 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Skeleton/TextSkeleton.tsx:45]: тут к сожалению есть проблема:  https://github.com/salute-developers/plasma/issues/230
поэтому пока так

File: packages/plasma-core/src/utils/extractTextFrom.ts (hunk 5 above)
Comment (Yeti-or) [reply packages/plasma-core/src/utils/extractTextFrom.ts:16]: да пожалуй

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-231/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-231/

  </details>

**Review by Yakutoc — APPROVED** [review]

**Pull request merged**